# **Project Name**    - *Super Stores Sales Analysis*

---





##### **Project Type**    - EDA

# **Project Summary -**

Write the summary here within 500-600 words.

# **GitHub Link -**

Provide your GitHub Link here.

# **Problem Statement**


**Write Problem Statement Here.**

#### **Define Your Business Objective?**

Answer Here.

# **General Guidelines** : -  

1.   Well-structured, formatted, and commented code is required.
2.   Exception Handling, Production Grade Code & Deployment Ready Code will be a plus. Those students will be awarded some additional credits.
     
     The additional credits will have advantages over other students during Star Student selection.
       
             [ Note: - Deployment Ready Code is defined as, the whole .ipynb notebook should be executable in one go
                       without a single error logged. ]

3.   Each and every logic should have proper comments.
4. You may add as many number of charts you want. Make Sure for each and every chart the following format should be answered.
        

```
# Chart visualization code
```
            

*   Why did you pick the specific chart?
*   What is/are the insight(s) found from the chart?
* Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

5. You have to create at least 20 logical & meaningful charts having important insights.


[ Hints : - Do the Vizualization in  a structured way while following "UBM" Rule.

U - Univariate Analysis,

B - Bivariate Analysis (Numerical - Categorical, Numerical - Numerical, Categorical - Categorical)

M - Multivariate Analysis
 ]





# ***Let's Begin !***

## ***1. Know Your Data***

### Import Libraries

In [ ]:
# Import Libraries
import numpy as np
import pandas as pd
import matplotlib.pyplot as plt
import seaborn as sns
import plotly.express as px

### Dataset Loading

In [ ]:
# Load Dataset
super_store_Sales_url = "https://docs.google.com/spreadsheets/d/1ZgMk5MV0rFiEGUNKgdkfNiywuvd1gRs7CNPNo-kmKRU/export?format=csv"

In [ ]:
sales = pd.read_csv(super_store_Sales_url)

### Dataset First View

In [ ]:
# Dataset First Look
sales.head()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
0,1,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-BO-10001798,Furniture,Bookcases,Bush Somerset Collection Bookcase,261.9600,2,0.00,41.9136
1,2,CA-2016-152156,11/8/2016,11/11/2016,Second Class,CG-12520,Claire Gute,Consumer,United States,Henderson,...,42420,South,FUR-CH-10000454,Furniture,Chairs,"Hon Deluxe Fabric Upholstered Stacking Chairs,...",731.9400,3,0.00,219.5820
2,3,CA-2016-138688,6/12/2016,6/16/2016,Second Class,DV-13045,Darrin Van Huff,Corporate,United States,Los Angeles,...,90036,West,OFF-LA-10000240,Office Supplies,Labels,Self-Adhesive Address Labels for Typewriters b...,14.6200,2,0.00,6.8714
3,4,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,FUR-TA-10000577,Furniture,Tables,Bretford CR4500 Series Slim Rectangular Table,957.5775,5,0.45,-383.0310
4,5,US-2015-108966,10/11/2015,10/18/2015,Standard Class,SO-20335,Sean O'Donnell,Consumer,United States,Fort Lauderdale,...,33311,South,OFF-ST-10000760,Office Supplies,Storage,Eldon Fold 'N Roll Cart System,22.3680,2,0.20,2.5164


In [ ]:
sales.tail()

,Row ID,Order ID,Order Date,Ship Date,Ship Mode,Customer ID,Customer Name,Segment,Country,City,...,Postal Code,Region,Product ID,Category,Sub-Category,Product Name,Sales,Quantity,Discount,Profit
9989,9990,CA-2014-110422,1/21/2014,1/23/2014,Second Class,TB-21400,Tom Boeckenhauer,Consumer,United States,Miami,...,33180,South,FUR-FU-10001889,Furniture,Furnishings,Ultra Door Pull Handle,25.248,3,0.2,4.1028
9990,9991,CA-2017-121258,2/26/2017,3/3/2017,Standard Class,DB-13060,Dave Brooks,Consumer,United States,Costa Mesa,...,92627,West,FUR-FU-10000747,Furniture,Furnishings,Tenex B1-RE Series Chair Mats for Low Pile Car...,91.960,2,0.0,15.6332
9991,9992,CA-2017-121258,2/26/2017,3/3/2017,Standard Class,DB-13060,Dave Brooks,Consumer,United States,Costa Mesa,...,92627,West,TEC-PH-10003645,Technology,Phones,Aastra 57i VoIP phone,258.576,2,0.2,19.3932
9992,9993,CA-2017-121258,2/26/2017,3/3/2017,Standard Class,DB-13060,Dave Brooks,Consumer,United States,Costa Mesa,...,92627,West,OFF-PA-10004041,Office Supplies,Paper,"It's Hot Message Books with Stickers, 2 3/4"" x 5""",29.600,4,0.0,13.3200
9993,9994,CA-2017-119914,5/4/2017,5/9/2017,Second Class,CC-12220,Chris Cortes,Consumer,United States,Westminster,...,92683,West,OFF-AP-10002684,Office Supplies,Appliances,"Acco 7-Outlet Masterpiece Power Center, Wihtou...",243.160,2,0.0,72.9480


### Dataset Rows & Columns count

In [ ]:
# Dataset Rows & Columns count
_shape = sales.shape
print(f'The sales Dataset has {_shape} Rows & Columns')

The sales Dataset has (9994, 21) Rows & Columns


### Dataset Information

In [ ]:
# Dataset Info
sales.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Row ID         9994 non-null   int64  
 1   Order ID       9994 non-null   object 
 2   Order Date     9994 non-null   object 
 3   Ship Date      9994 non-null   object 
 4   Ship Mode      9994 non-null   object 
 5   Customer ID    9994 non-null   object 
 6   Customer Name  9994 non-null   object 
 7   Segment        9994 non-null   object 
 8   Country        9994 non-null   object 
 9   City           9994 non-null   object 
 10  State          9994 non-null   object 
 11  Postal Code    9994 non-null   int64  
 12  Region         9994 non-null   object 
 13  Product ID     9994 non-null   object 
 14  Category       9994 non-null   object 
 15  Sub-Category   9994 non-null   object 
 16  Product Name   9994 non-null   object 
 17  Sales          9994 non-null   float64
 18  Quantity

### Duplicate Values

In [ ]:
# Dataset Duplicate Value Count
_duplicates = sales.duplicated().sum()
print(f'the sales Dataset has {_duplicates} duplicate values')

the sales Dataset has 0 duplicate values


### Missing Values/Null Values

In [ ]:
# Missing Values/Null Values Count
print(sales.isnull().sum())
print("sales has 0 Null Values")

Row ID           0
Order ID         0
Order Date       0
Ship Date        0
Ship Mode        0
Customer ID      0
Customer Name    0
Segment          0
Country          0
City             0
State            0
Postal Code      0
Region           0
Product ID       0
Category         0
Sub-Category     0
Product Name     0
Sales            0
Quantity         0
Discount         0
Profit           0
dtype: int64
sales has 0 Null Values


In [ ]:
# Visualizing the missing values
## THERE'S NO NULL VALUES TO VISUALIZE

### What did you know about your dataset?

*  The "sales" dataset has a description of a xyz superstore's Sales
*  The "sales" dataset has 9994 Rows and 21 Columns in it.
*  The "sales" dataset has Zero missing/Null Values.
*  The "sales" dataset has Zero Duplicate Values.

## ***2. Understanding Your Variables***

In [ ]:
# Dataset Columns
sales_columns = sales.columns
print("The sales data has these columns :- ")
print(sales_columns)

The sales data has these columns :- 
Index(['Row ID', 'Order ID', 'Order Date', 'Ship Date', 'Ship Mode',
       'Customer ID', 'Customer Name', 'Segment', 'Country', 'City', 'State',
       'Postal Code', 'Region', 'Product ID', 'Category', 'Sub-Category',
       'Product Name', 'Sales', 'Quantity', 'Discount', 'Profit'],
      dtype='object')


In [ ]:
# Dataset Describe
_description = sales.describe()
print('The Description of dataset is :- ')
print(_description)

The Description of dataset is :- 
            Row ID   Postal Code         Sales     Quantity     Discount  \
count  9994.000000   9994.000000   9994.000000  9994.000000  9994.000000   
mean   4997.500000  55190.379428    229.858001     3.789574     0.156203   
std    2885.163629  32063.693350    623.245101     2.225110     0.206452   
min       1.000000   1040.000000      0.444000     1.000000     0.000000   
25%    2499.250000  23223.000000     17.280000     2.000000     0.000000   
50%    4997.500000  56430.500000     54.490000     3.000000     0.200000   
75%    7495.750000  90008.000000    209.940000     5.000000     0.200000   
max    9994.000000  99301.000000  22638.480000    14.000000     0.800000   

            Profit  
count  9994.000000  
mean     28.656896  
std     234.260108  
min   -6599.978000  
25%       1.728750  
50%       8.666500  
75%      29.364000  
max    8399.976000  


### Variables Description

**Order Information**

Row ID : Unique identifier for each record/transaction row.

Order ID : Unique identifier assigned to a customer order.

Order Date : Date on which the order was placed.

Ship Date : Date on which the order was shipped.

Ship Mode : Shipping method used for delivery (e.g., Standard Class, Second Class, First Class, Same Day).

**Customer Information**

Customer ID : Unique identifier assigned to a customer.

Customer Name : Name of the customer who placed the order.

Segment : Customer category or market segment (Consumer, Corporate, Home Office).

**Location Information**

Country : Country where the order was placed.

City : City associated with the order.

State : State associated with the order.

Postal Code : Postal/ZIP code of the customer's location.

Region : Geographic sales region (East, West, Central, South).

**Product Information**

Product ID : Unique identifier assigned to a product.

Category : High-level product category (Furniture, Office Supplies, Technology).

Sub-Category : More specific classification of products within a category.

Product Name : Name or description of the product sold.

**Sales & Performance Metrics**

Sales : Revenue generated from the sale of the product.

Quantity : Number of units sold in the transaction.

Discount : Discount applied to the product (represented as a proportion).

Profit : Profit or loss earned from the transaction after accounting for costs and discounts.

### Check Unique Values for each variable.

In [ ]:
# Checking Unique Values for each variable.
# ------------------------------------------------------------


print("Unique Value Analysis for sales Dataset")
print("-" * 50)

for col in sales.columns:
    unique_count = sales[col].nunique()
    print(f"Column: {col:<25} | Unique Values: {unique_count}")

print("\n" + "-" * 50)

Unique Value Analysis for sales Dataset
--------------------------------------------------
Column: Row ID                    | Unique Values: 9994
Column: Order ID                  | Unique Values: 5009
Column: Order Date                | Unique Values: 1237
Column: Ship Date                 | Unique Values: 1334
Column: Ship Mode                 | Unique Values: 4
Column: Customer ID               | Unique Values: 793
Column: Customer Name             | Unique Values: 793
Column: Segment                   | Unique Values: 3
Column: Country                   | Unique Values: 1
Column: City                      | Unique Values: 531
Column: State                     | Unique Values: 49
Column: Postal Code               | Unique Values: 631
Column: Region                    | Unique Values: 4
Column: Product ID                | Unique Values: 1862
Column: Category                  | Unique Values: 3
Column: Sub-Category              | Unique Values: 17
Column: Product Name              | 

## 3. ***Data Wrangling***

### Data Wrangling Code

In [ ]:
# Write your code to make your dataset analysis ready.
# we do not have any  null and duplicate values so we will write code for {ibject to datetime} conversion

sales.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 9994 entries, 0 to 9993
Data columns (total 21 columns):
 #   Column         Non-Null Count  Dtype  
---  ------         --------------  -----  
 0   Row ID         9994 non-null   int64  
 1   Order ID       9994 non-null   object 
 2   Order Date     9994 non-null   object 
 3   Ship Date      9994 non-null   object 
 4   Ship Mode      9994 non-null   object 
 5   Customer ID    9994 non-null   object 
 6   Customer Name  9994 non-null   object 
 7   Segment        9994 non-null   object 
 8   Country        9994 non-null   object 
 9   City           9994 non-null   object 
 10  State          9994 non-null   object 
 11  Postal Code    9994 non-null   int64  
 12  Region         9994 non-null   object 
 13  Product ID     9994 non-null   object 
 14  Category       9994 non-null   object 
 15  Sub-Category   9994 non-null   object 
 16  Product Name   9994 non-null   object 
 17  Sales          9994 non-null   float64
 18  Quantity

In [ ]:
# converting 'Order Date' column to "datetime"
sales['Order Date'] = pd.to_datetime(sales['Order Date'])

In [ ]:
# converting 'Order Date' column to "datetime"
sales['Ship Date'] = pd.to_datetime(sales['Ship Date'])

### What all manipulations have you done and insights you found?

####**Manipulations :-**

In the data warngling step, i have first run the "sales.info()" column.

in that i have found that my dataset do not have any null value,

so after date i search for any columns that have values in datetime format but it is in object datatype,

then in got 2 columns that were in the same condition, so i converted them into a datetime

****
####**Insights :-**


*   If there is no null Values to fill then we can leave the data wrangling process OR we can go to other columns that have Wrong Dtype
*   It is necessary to convert the date columns to datetime Dtype





In [ ]:
# Adding a column named year_month
sales['year_month'] = sales['Order Date'].dt.to_period('M')

## 4. ***Questions Related to Dataset***

####which category is generating most revenue

In [ ]:
salas_cat = sales.groupby('Category')['Sales'].sum()
salas_cat

,Sales
Category,
Furniture,741999.7953
Office Supplies,719047.0320
Technology,836154.0330


####Which product is selling the most?

In [ ]:
_most_sale_sub_cat = sales['Sub-Category'].value_counts().head(5)
print("The top selling sub_categories are :- \n", _most_sale_sub_cat)


The top selling sub_categories are :- 
 Sub-Category
Binders        1523
Paper          1370
Furnishings     957
Phones          889
Storage         846
Name: count, dtype: int64


In [ ]:
px.histogram(sales, 'Sub-Category')

####Which city generated the most revenue?

In [ ]:
# Assuming your DataFrame is named 'superstore_sales'
Most_sales_city = sales.groupby('City')['Sales'].sum().reset_index().sort_values(by='Sales', ascending=False).head(1)
print("The City Generates the most revenue is :- \n", Most_sales_city)


The City Generates the most revenue is :- 
               City       Sales
329  New York City  256368.161


####profit VS sales analysis

In [ ]:
sales_sorted = sales.sort_values(by='Sales', ascending=True)
pVs = px.line(sales_sorted, x='Sales', y='Profit')
pVs.show()

####Discount VS Profit

In [ ]:
_fig = px.line(sales, 'Profit', 'Discount')
_fig.show()

####sales analysis by month

In [ ]:
# sns.barplot(data=sales, x= 'year_month', y= 'Sales')
# plt.title("Sales Analysis by Month")
# plt.xlabel("Order month")
# plt.ylabel("Sales")
# plt.show()
sales['year_month'] = sales['year_month'].astype(str)
_fig = px.bar(sales, x= 'year_month', y='Sales', title='Sales Analysis by Month')
_fig.show()

####Profit by Month

In [ ]:
monthly_data = sales.groupby('year_month')['Profit'].sum().reset_index()

In [ ]:
PbM = px.bar(monthly_data, 'Profit')
PbM.show()

####Year-wise Monthly Profit Trend

In [ ]:
monthly_data['year_month'] = monthly_data['year_month'].astype(str)

fig = px.line(monthly_data, x='year_month', y='Profit', title='Year-wise Monthly Profit Trend')
fig.show()

## ***5. Data Vizualization, Storytelling & Experimenting with charts : Understand the relationships between variables***

#### Chart - 1

In [ ]:
# Chart - 1 visualization code


##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 2

In [ ]:
# Chart - 2 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 3

In [ ]:
# Chart - 3 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 4

In [ ]:
# Chart - 4 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 5

In [ ]:
# Chart - 5 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 6

In [ ]:
# Chart - 6 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 7

In [ ]:
# Chart - 7 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 8

In [ ]:
# Chart - 8 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 9

In [ ]:
# Chart - 9 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 10

In [ ]:
# Chart - 10 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 11

In [ ]:
# Chart - 11 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 12

In [ ]:
# Chart - 12 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 13

In [ ]:
# Chart - 13 visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

##### 3. Will the gained insights help creating a positive business impact?
Are there any insights that lead to negative growth? Justify with specific reason.

Answer Here

#### Chart - 14 - Correlation Heatmap

In [ ]:
# Correlation Heatmap visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

#### Chart - 15 - Pair Plot

In [ ]:
# Pair Plot visualization code

##### 1. Why did you pick the specific chart?

Answer Here.

##### 2. What is/are the insight(s) found from the chart?

Answer Here

## **6. Solution to Business Objective**

#### What do you suggest the client to achieve Business Objective ?
Explain Briefly.

Answer Here.

# **Conclusion**

Write the conclusion here.

### ***Hurrah! You have successfully completed your EDA Capstone Project !!!***